src/classifier_sequence.py

```diff
@@ -0,0 +1,100 @@
+import pathlib
+import math
+import numpy as np
+from typing import Dict, Tuple, List
+from tensorflow import keras
+from PIL import Image, ImageOps
+
+
+class ClassifierSequence(keras.utils.Sequence):
+    """A class for sequencing the classifier's input data."""
+
+    category_id_to_name: Dict[int, str]
+    category_name_to_id: Dict[str, int]
+
+    def __init__(self, input_dir: pathlib.Path, batch_size: int,
+                 target_img_dim: int, shuffle: bool = True):
+        """Constructor.
+
+        :param input_dir: directory with input data
+        :param batch_size: the desired size of a batch
+        :param target_img_dim: target dimension (for square icons)
+        :param shuffle: whether the input data is shuffled on epoch end or not.
+        """
+        if not input_dir.exists():
+            raise ValueError("Input directory does not exist.")
+
+        self._input_dir = input_dir
+        self._batch_size = batch_size
+        self._app_icons = []
+        self._shuffle = shuffle
+        self._target_icon_size = (target_img_dim, target_img_dim)
+
+        categories = self._get_categories()
+
+        self.category_id_to_name = dict(enumerate(categories))
```

Comment:
Could you add typehints as discussed?

Reply:
The typehints were added before the constructor. Should I move them in here?